github/client_repository_file.go

```diff
@@ -54,6 +74,35 @@ func (c *FileClient) Get(ctx context.Context, path, branch string) ([]*gitprovid
 
 	for _, file := range directoryContent {
 		filePath := file.Path
+		if *file.Type == "dir" {
+			if fileOpts.Recursive {
+				// stop recursive calls when level is the max level reached
+				if fileOpts.MaxDepth <= 1 {
+					continue
+				}
+
+				if !strings.HasSuffix(path, "/") {
+					path = path + "/"
+				}
+				subdirectoryPath := fmt.Sprintf("%s%s/", path, *file.Name)
+
+				// recursive call for child directories to get their content
+				childMaxFiles := fileOpts.MaxFiles - len(files)
+				childOptFns := gitprovider.FilesGetOptions{Recursive: fileOpts.Recursive, MaxFiles: childMaxFiles, MaxDepth: fileOpts.MaxDepth - 1}
+				childFiles, err := c.Get(ctx, subdirectoryPath, branch, &childOptFns)
+
+				if err != nil {
+					return nil, err
+				}
+				files = append(files, childFiles...)
+				files, limitSuccess := limitSliceSize(files, fileOpts.MaxFiles)
```

Comment:
can we rename `limitSuccess` to something that match a boolean `shoudReturn` or something

Reply:
will change it to `maxFilesReached`